rfcs/0051-close-stale-issues.md

```diff
@@ -0,0 +1,74 @@
+---
+feature: close-stale-issues
+start-date: 2019-08-24
+author: Ryan Mulligan
+co-authors: (find a buddy later to help our with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Close stale Nixpkgs issues and pull requests (hereafter both referred
+to as simply "issues") on GitHub using an application provided by
+GitHub.
+
+# Motivation
+[motivation]: #motivation
+
+The Nixpkgs GitHub page has a large number of open issues causing
+community angst and misrepresenting the responsiveness of the project.
+
+By closing stale issues, we can (in an automated way) refocus our
+efforts on the ones that have at least one person interested in them.
+
+# Detailed design
+[design]: #detailed-design
+
+1. Use the [Stale](https://github.com/marketplace/stale) application
+   provided by GitHub on the [Nixpkgs
+   repository](https://github.com/NixOS/nixpkgs).
+2. Start by using the following `.github/stale.yml` configuration
+   file:
+
+   ```
+   # Number of days of inactivity before an issue becomes stale
+   daysUntilStale: 60
+   # Number of days of inactivity before a stale issue is closed
+   daysUntilClose: 7
+   # Issues with these labels will never be considered stale
+   exemptLabels:
+   # Label to use when marking an issue as stale
+   staleLabel: stale
+   # Comment to post when marking an issue as stale. Set to `false` to disable
+   markComment: >
+     This issue has been automatically marked as stale because it has not had
+     recent activity. It will be closed if no further activity occurs. Thank you
+     for your contributions.
```

Comment:
I think the framing here is very important to (1) don't make anyone feel like they're just a drag on the project, make clear that it's totally fine to just keep bumping an issue; (2) give constructive feedback on how to get attention for an issue, like actively searching for maintainers / previous people that touched the code and pinging them.

Reply:
I think this is a great point. Phrasing it as helping the user seems much more likely to turn out well. "It looks like this issue hasn't attracted much attention. Here are n suggestions for how to progress this issue further."